src/NetscapeBookmarkParser.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shaarli\NetscapeBookmarkParser;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Shaarli\NetscapeBookmarkParser\DataFormatter\BookmarksDataFormatter;
+use Shaarli\NetscapeBookmarkParser\Encoder\NetscapeBookmarkDecoder;
+use Shaarli\NetscapeBookmarkParser\Encoder\NetscapeBookmarkEncoder;
+
+/**
+ * Generic Netscape bookmark parser
+ */
+class NetscapeBookmarkParser
+{
+    public const DEPRECATION_VERSION = '4.0';
+
+    private $defaultContext = [
+        NetscapeBookmarkDecoder::KEEP_NESTED_TAGS => true,
+        NetscapeBookmarkDecoder::NORMALIZE_DATES  => true,
+        NetscapeBookmarkDecoder::DATE_RANGE       => '30 years',
+    ];
+
+    /**
+     * @var LoggerInterface instance.
+     */
+    private $logger;
+
+    /**
+     * @var NetscapeBookmarkDecoder
+     */
+    private $decoder;
+
+    /**
+     * @var BookmarksDataFormatter
+     */
+    private $formatter;
+
+    /**
+     * @var NetscapeBookmarkEncoder
+     */
+    private $encoder;
+
+    public function __construct(array $defaultContext = [], LoggerInterface $logger = null)
+    {
+        $this->defaultContext = array_merge($this->defaultContext, $defaultContext);
+        $this->logger = $logger ?? new NullLogger();
+
+        $this->decoder = new NetscapeBookmarkDecoder($this->defaultContext, $this->logger);
+        $this->formatter = new BookmarksDataFormatter();
+        $this->encoder = new NetscapeBookmarkEncoder();
+    }
+
+    /**
+     * Parses a Netscape bookmark file
+     *
+     * @param string $filePath Bookmark file to parse
+     *
+     * @return array An associative array containing parsed links
+     */
+    public function parseFile(string $filePath): array
+    {
+        $this->logger->info('Starting to parse ' . $filePath);
+
+        try {
+            return $this->decoder->decode(file_get_contents($filePath));
+        } catch (\Exception $exception) {
+            $this->logger->error($exception);
+        }
+
+        return [];
+    }
+
+    /**
+     * Exports data to Netscape bookmark format file
+     *
+     * @param array  $data
+     * @param string $filePath
+     *
+     * @return string A string containing serialized data
+     */
+    public function export(array $data, string $filePath): string
+    {
+        $this->logger->info('Export to ' . $filePath);
+
+        $data = $this->formatter->format($data);
+
+
+        try {
+            $data = $this->encoder->encode($data);
+            var_dump($data);
+            die();
+        } catch (\Exception $exception) {
+            $this->logger->error($exception);
+        }
+
+        file_put_contents($filePath, $data);
+
+        return $data;
+    }
+
+    /**
+     * Parses a string containing Netscape-formatted bookmarks
+     *
+     * @param string $data
+     *
+     * @return array
+     * @deprecated
+     */
+    public function parseString(string $data): array
```

Comment:
We should keep this one. Sometimes we don't want to deal with files and just manipulate strings.

Reply:
Well in that case, shouldn't the client use  `decoder->decode()` directly ?

Reply:
The library should have a single entry point (class) with necessary public methods, even if they're only wrappers.

Reply:
I assumed that `NetscapeBookmarkParser` would be removed at some point. I believe it is better for users to implement their use case with new `encoder` and `decoder` directly.